test/open_api/map_test.exs

```diff
@@ -0,0 +1,403 @@
+    test "OpenApi" do
+      # NOTE: This test could be split into many smaller tests, that is the goal!
+      # TODO: Move the spec to a setup below, where we could easily try a yaml version also
+      spec =
+        "./test/support/encoded_schema.json"
+        |> File.read!()
+        |> Jason.decode!(keys: :atoms)
+        |> OpenApiSpex.OpenApi.Map.decode()
+
+      # OpenAPI Object
+      assert %OpenApi{
+               openapi: openapi,
+               info: info,
+               servers: servers,
+               paths: paths,
+               components: components,
+               security: security,
+               tags: tags,
+               externalDocs: externalDocs,
+               extensions: extensions
+             } = spec
+
+      assert "3.0.0" == openapi
+
+      # OpenApi Info Object
+      assert %OpenApiSpex.Info{
+               title: _title,
+               version: _version,
+               description: _description,
+               contact: contact,
+               license: license
+             } = info
+
+      # OpenApi Contact Object
+      assert %OpenApiSpex.Contact{} = contact
+
+      # OpenApi Contact Object
+      assert %OpenApiSpex.License{} = license
+
+      # OpenApi Specification Extensions
+      assert %{
+               test: true
+             } == extensions
+
+      # OpenApi External Documentation Object
+      assert %OpenApiSpex.ExternalDocumentation{
+               description: _,
+               url: _
+             } = externalDocs
+
+      # OpenApi Components Object
+      assert %OpenApiSpex.Components{
+               callbacks: callbacks,
+               schemas: schemas,
+               responses: responses,
+               examples: examples,
+               links: %{
+                 "address" => link
+               },
+               requestBodies: requestBodies,
+               parameters: %{
+                 "AcceptEncodingHeader" => components_parameters_parameter
+               },
+               securitySchemes: securitySchemes,
+               headers: %{
+                 "api-version" => components_headers_header
+               }
+             } = components
+
+      assert %{
+               "test" => %OpenApiSpex.RequestBody{
+                 description: "user to add to the system",
+                 content: %{
+                   "application/json": media_type
+                 },
+                 required: false
+               }
+             } = requestBodies
+
+      assert %OpenApiSpex.MediaType{
+               schema: %OpenApiSpex.Reference{},
+               examples: %{
+                 user: %OpenApiSpex.Example{}
+               },
+               encoding: %{
+                 historyMetadata: %OpenApiSpex.Encoding{
+                   contentType: "application/xml; charset=utf-8",
+                   style: :simple,
+                   explode: false,
+                   allowReserved: false,
+                   headers: %{
```

Comment:
`:headers` should be a map with string keys: https://github.com/GhostGroup/open_api_spex/blob/daf46b32f87c24c69a16859077e6385472966f70/lib/open_api_spex/encoding.ex#L21